src/ServiceControl/Monitoring/HeartbeatMonitoringComponent.cs

```diff
@@ -21,7 +21,7 @@ public override void Configure(Settings settings, IHostBuilder hostBuilder)
                 collection.AddSingleton<IDataMigration, PurgeKnownEndpointsWithTemporaryIdsThatAreDuplicateDataMigration>();
                 collection.AddHostedService<HeartbeatMonitoringHostedService>();
                 collection.AddSingleton<EndpointInstanceMonitoring>();
-                collection.AddSingleton<MonitoringDataStore>();
+                collection.AddSingleton<IMonitoringDataStore, SqlDbMonitoringDataStore>();
```

Comment:
Is this perhaps where we could have a setting on `Settings` to specify the data store for heartbeats? Based on that setting we could then change what we load into the collection.